ProcessManager.py

```diff
@@ -79,6 +82,24 @@ def isEmpty(self):
         """Return if processes is empty."""
         return len(self.processes) == 0
 
+    def loadConfig(self, file_name):
+        """Loads process configuration from a file"""
+        if not isinstance(file_name, str):
+            raise TypeError("Path to config file must be a string")
+        with open(file_name) as file:
+            config = json.load(file)
+            local_processes = config['localhost']
```

Comment:
change the key to [`socket.gethostname()`](https://docs.python.org/3/library/socket.html?highlight=socket%20bind#socket.gethostname)? It would allow for a single config file for multiple hosts.

Reply:
That would work. My original thinking was that the "master" node would be the only one to run with the config file, and all the other hosts would just be waiting to be told what processes to run. The master node could also send the most up to date versions of programs to the other nodes.
If I used `socket.gethostname()`, would there be an easy way to write the config file to have an entry for _this_ computer that is running? In the case of testing for example, my laptop's hostname is `Archpad`, but if I put a config entry for `Archpad`, that process group woudn't run on anyone elses machine without modification, possibly making automated testing difficult.

Reply:
Could we maybe run any hosts that it cannot find on the localhost?

Reply:
That is an option, my only worry about it is that if a host is down, it is likely it will have devices physically attached to it that the processes require to function properly. Running those processes on the localhost would let the system start properly, but the system wouldn't necessarily work properly.

How about this: The manger checks for both the `localhost` and `socket.gethostname()` fields and runs both those process sets. That allows the config files to be portable between machines, and allows you to assure that certain processes are run on the localhost.

Reply:
Checking localhost and hostname is probably the cleanest way